dist/src/main/java/io/camunda/zeebe/broker/WorkingDirectoryConfiguration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker;
+
+import io.camunda.zeebe.shared.Profile;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.Profiles;
+import org.springframework.web.context.annotation.ApplicationScope;
+
+@Configuration(proxyBeanMethods = false)
+public record WorkingDirectoryConfiguration(Environment environment) {
+
+  @Bean
+  @ApplicationScope(proxyMode = ScopedProxyMode.NO)
```

Comment:
:wrench: Default proxyMode = No. So may be it is unnecessary set it everywhere.

Reply:
Is it? I had issues where it tried to proxy it when setting the scope without it, on final classes :thinking: Let me try again

Reply:
so the default `proxyMode` is `ScopedProxyMode.TARGET_CLASS` for me (when using `ApplicationScope` anyway).

I talked with Ole before about not needing this scope, since the default is the singleton scope, but I noticed while debugging that without it, the method was being called multiple time. I didn't dig too much into it, but maybe this is the wrong approach as well :thinking:

Reply:
OK, I figured out why some beans are called multiple times :smile: Spring will eager initialize the beans, even if no one is requesting them for auto wiring. Since I also sometimes call the methods directly, this was causing problems. So the issue is, don't ever call bean methods directly :smile: